tests/Agent/UnitTests/Core.UnitTest/ThreadProfiling/ThreadProfilingSamplerTests.cs

```diff
@@ -0,0 +1,143 @@
+// Copyright 2020 New Relic, Inc. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+using System;
+using NUnit.Framework;
+using Telerik.JustMock;
+using System.Threading;
+using System.Threading.Tasks;
+using System.Runtime.InteropServices;
+
+namespace NewRelic.Agent.Core.ThreadProfiling
+{
+    [TestFixture]
+    public class ThreadProfilingSamplerTests
+    {
+        private INativeMethods _nativeMethods;
+        private ThreadProfilingSampler _threadProfiler;
+        private ISampleSink _sampleSink;
+        private ManualResetEventSlim _shutdownEvent;
+
+        [SetUp]
+        public void Setup()
+        {
+            _nativeMethods = Mock.Create<INativeMethods>();
+            _sampleSink = Mock.Create<ISampleSink>();
+            _shutdownEvent = Mock.Create<ManualResetEventSlim>();
```

Comment:
Is this something that needs to be injected into the test? It doesn't look like any of the tests use it directly.

Reply:
Oops. Leftover from some early work I was doing where I thought I needed to mock the ManualResetEventSlim. I'll remove it (and the change in `ThreadProfilingSampler`)